src/builder-main.c

```diff
@@ -1097,9 +1097,13 @@ main (int    argc,
                                        builder_context_get_build_runtime (build_context) ? "usr" : "files",
                                        builder_extension_get_directory (e));
 
+          const char *extension_branch = builder_extension_get_version(e);
+          if (extension_branch == NULL)
+            extension_branch = builder_manifest_get_branch (manifest, build_context);
+
           if (!do_export (build_context, &error, TRUE,
                           flatpak_file_get_path_cached (export_repo),
-                          app_dir_path, NULL, builder_manifest_get_branch (manifest, build_context),
+                          app_dir_path, NULL, extension_branch,
```

Comment:
Do we need a similar change in `builder_manifest_finish()` since that also iterates over the non-bundled extensions and writes metadata for them? Have you checked if this patch works for fixing #388?

Reply:
>Do we need a similar change in `builder_manifest_finish()` since that also iterates over the non-bundled extensions and writes metadata for them?

No, unless I'm missing something, `builder_manifest_finish()` doesn't write the extension branch anywhere.

>Have you checked if this patch works for fixing #388?

Yes, although for another case. It seems to do the trick.

Reply:
> > Do we need a similar change in `builder_manifest_finish()` since that also iterates over the non-bundled extensions and writes metadata for them?
> 
> No, unless I'm missing something, `builder_manifest_finish()` doesn't write the extension branch anywhere.

Yeah, I see now that the `version` key can be defined for `[Extension NAME]` sections according to `flatpak-metadata(5)` but not `[Runtime]` as is being written in this case. So I guess you're right, it's not needed.